### PR TITLE
Implement server-side processing of datatables in users#index

### PR DESCRIPTION
**Pull request by nishanthvijayan**
Solves #581 
-  I've not implemented sorting by Attended Conferences
-  Search checks for matches only in name and email fields

### CODE REVIEW

**Comment by differentreality** [comment]
@hennevogel do you have any input about this?

File: app/controllers/concerns/datatable_serverside_processing.rb [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  end
+
+  private
+
+  def records
+    @records ||= fetch_records
+  end
+
+  def fetch_records
+    sort_direction = params[:sSortDir_0] == 'desc' ? 'desc' : 'asc'
```
Comment (ChrisBr) [inline app/controllers/concerns/datatable_serverside_processing.rb]: move this to a dedicated function.

```
def sort_direction
...
```

File: app/controllers/concerns/datatable_serverside_processing.rb [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    sort_column = @sort_columns[params[:iSortCol_0].to_i]
```
Comment (ChrisBr) [inline app/controllers/concerns/datatable_serverside_processing.rb]: This also

``` 
def sort_column
...
```

File: app/controllers/concerns/datatable_serverside_processing.rb [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    records = @collection.order("#{sort_column} #{sort_direction}")
+
+    if params[:sSearch].present? && @search_columns.present?
```
Comment (ChrisBr) [inline app/controllers/concerns/datatable_serverside_processing.rb]: Move this also to a dedicated function!

**Comment by differentreality** [comment]
@mdeniz did you have a chance to take a look? Any thoughts?

**Comment by Ana06** [comment]
@differentreality @mdeniz @nishanthvijayan what is the state of this? 🤔

**Comment by differentreality** [comment]
Huge change, but I do not know if there is another way.
We do need the feature one way or the other.

@mdeniz had mentioned at some point he can have a look, but review is still pending.

I don't know if perhaps @bear454 is also up for this.

**Comment by bear454** [comment]
@nishanthvijayan: I pulled down this branch, and tried it against master, it doesn't work for me. I spent a little time diving into making changes, and I just don't see an easy route forward without rearchitecting.

IMO the best way forward would be to implement https://github.com/jbox-web/ajax-datatables-rails instead of reinventing the wheel.

**Comment by hennevogel** [comment]
I agree with @bear454, let's close this. Sorry @nishanthvijayan

**Pull request closed without merging**